cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,231 @@
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    def product_name(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Product Name")
+
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    def health_state(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Health State")
+
+    def enclosure_count(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Enclosure Count")
+
+    def drive_bays(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Drive Bays")
+
+    def uuid(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("UUID")
+
+    def description(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Description")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for the Physical Storage to reach the desired state. This function waits an amount
+           of time due to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            provider (object): 'LenovoProvider'
+            view (object): The view that most be refreshed to verify if the value was changed
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
+            return desired_state == getattr(self, target)()
+
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_Storage exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def db_id(self):
+        return self.appliance.physical_storage_id(self.name)
+
+    def wait_to_appear(self):
```

Comment:
Required:
`wait_to_appear` and `wait_for_delete` both are depends on `exists` properly better to handle those in test only.